compiler/rustc_codegen_llvm/src/llvm/ffi.rs

```diff
@@ -980,8 +982,10 @@ pub(crate) unsafe fn enzyme_rust_reverse_diff(
     let mut known_values = vec![kv_tmp; input_tts.len()];
 
     let dummy_type = CFnTypeInfo {
-        Arguments: args_tree.as_mut_ptr(),
-        Return: output_tt.inner.clone(),
+        Arguments: std::ptr::null_mut(),
```

Comment:
@wsmoses What does Enzyme expect here?
Passing in nullpointer (or for the arguments an array of nullpointers causes a crash. 
However, if I set the return to an empty tt (and the input to an array of empty tt), then I ran into an Enzyme core assertion (unfortunately not getting which one precisely due to ffi boundaries). What is it supposed to be now? I would (strongly) prefer to not pass tt information in two ways.

Reply:
I’d presume empty TT, what’s the error?

youd be the first to pass in null TT always (Julia still prepopulates the outermost one and uses the function attrs for the interprocedural TA speed ups). I could imagine that the outer one isn’t getting populated from the attributes which should be a simple fix if that’s the error.

Reply:
Even under gdb I'm not getting full errors. However, it's in the 

Also, empty trees used to work in oxide-enzyme => https://github.com/EnzymeAD/oxide-enzyme/blob/ae3115d1e1d966a290324feeb04990ea9653be7e/src/enzyme/enzyme_wrapper.rs#L147
So maybe it regressed since I was using it, since for Rust-Enzyme I passed tt for most of the time, so it ended up untested?

Reply:
And just fyi, that's how the module now looks like, with ffff getting diffed
```
       1 ; Function Attrs: noinline nonlazybind sanitize_hwaddress uwtable
  9480   define dso_local "enzyme_type"="{}" void @ffff(ptr align 8 "enzyme_type"="{[-1]:Pointer, [-1,0]:Integer, [-1,1]:Integer, [-1,2]:Integer, [-1,3]:Integer, [-1,4]:Integer,          [-1,5]:Integer, [-1,6]:Integer, [-1,7]:Integer, [-1,8]:Pointer, [-1,8,-1]:Float@double, [-1,16]:Integer, [-1,17]:Integer, [-1,18]:Integer, [-1,19]:Integer, [-1,20]:Integ         er, [-1,21]:Integer, [-1,22]:Integer, [-1,23]:Integer}" %0, ptr align 8 "enzyme_type"="{[-1]:Pointer, [-1,-1]:Float@double}" %1) unnamed_addr #14 personality ptr @rust_e         h_personality !dbg !1120 {
       1   %3 = alloca { ptr, i32 }, align 8
```